streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueSegment.java

```diff
@@ -45,10 +46,9 @@ public int compareTo(final KeyValueSegment segment) {
     }
 
     @Override
-    public void openDB(final ProcessorContext context) {
-        super.openDB(context);
+    public void openDB(final Map<String, Object> configs, final File stateDir) {
+        super.openDB(configs, stateDir);
```

Comment:
I was able to remove the type-dependency of the context by re-specifying the interface in terms of the only two properties it needed.